shapely/geometry/base.py

```diff
@@ -182,10 +182,8 @@ def __ne__(self, other):
     @property
     def coords(self):
         """Access to geometry's coordinates (CoordinateSequence)"""
-        if self._coords is None:
-            coords_array = pygeos.get_coordinates(self, include_z=self.has_z)
-            self._coords = CoordinateSequence(coords_array)
-        return self._coords
+        coords_array = pygeos.get_coordinates(self, include_z=self.has_z)
+        return CoordinateSequence(coords_array)
```

Comment:
We can see later if we want to "cache" this again (there might be other attributes you could cache as well)